### PR TITLE
task_DEV-6

### PR DESCRIPTION
**Pull request by VadimBelkin**

### CODE REVIEW

File: IncreasingSequence.java [hunk 1]
```diff
@@ -1,38 +1,11 @@
...
-    } else {
-      Scanner scan = new Scanner(System.in);
-      System.out.print("Enter the numbers through a space: ");
-      String sequence = scan.nextLine();
-      String sequenceParts[] = sequence.split(" ");
-      nondecreasing(sequenceParts);
-    }
+    ReadingFromTheConsole readingTheSequence = new ReadingFromTheConsole();
+    SequenceAnalysis sequence = new SequenceAnalysis();
+      if (args.length != 0) {
```
Comment (AlexanderSoroka) [inline IncreasingSequence.java]: indentation is wrong

File: IncreasingSequence.java [hunk 2]
```diff
@@ -1,38 +1,11 @@
...
-    if (args.length != 0) {
-      nondecreasing(args);
-    } else {
-      Scanner scan = new Scanner(System.in);
-      System.out.print("Enter the numbers through a space: ");
-      String sequence = scan.nextLine();
-      String sequenceParts[] = sequence.split(" ");
-      nondecreasing(sequenceParts);
-    }
+    ReadingFromTheConsole readingTheSequence = new ReadingFromTheConsole();
```
Comment (AlexanderSoroka) [inline IncreasingSequence.java]: create object as close to first usage as possible

File: SequenceAnalysis.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * Analyzes the entered sequence.
+ */
+public class SequenceAnalysis {
+  /**
+   * Determines which sequence and displays information about this on the screen.
+   *
+   * @param sequence a user-defined sequence of numbers.
+   */
+  public void sequenceCheck(String[] seq) {
+    try{
```
Comment (AlexanderSoroka) [inline SequenceAnalysis.java]: codestyle violation

File: SequenceAnalysis.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      for (int i = 1; i < seq.length; i++) {   
+        if (Integer.parseInt(seq[i]) < Integer.parseInt(seq[i - 1])) {
+          System.out.println("Your numerical sequence is not nondecreasing.");
```
Comment (AlexanderSoroka) [inline SequenceAnalysis.java]: Single responsibility principle violation - the method must only check, not check and output results

File: IncreasingSequence.java [hunk 5]
```diff
@@ -1,38 +1,28 @@
...
-    } else {
-      Scanner scan = new Scanner(System.in);
-      System.out.print("Enter the numbers through a space: ");
-      String sequence = scan.nextLine();
-      String sequenceParts[] = sequence.split(" ");
-      nondecreasing(sequenceParts);
-    }
+    try {
+      SequenceAnalysis sequence = new SequenceAnalysis();
+        if (args.length != 0) {
```
Comment (AlexanderSoroka) [inline IncreasingSequence.java]: indentation is wrong

File: IncreasingSequence.java [hunk 6]
```diff
@@ -1,38 +1,28 @@
...
-   * <p>
-   * Determines which sequence and displays information about 
-   * this on the screen.
+  /**
+   * displays on the screen message about the kind of the sequence.
    *
-   * @param sequence a user-defined sequence of numbers.
+   * @param flag says whether our sequence increases or not.
    */
-  public static void nondecreasing(String[] sequence) {
```
Comment (AlexanderSoroka) [inline IncreasingSequence.java]: why static?

File: IncreasingSequence.java [hunk 7]
```diff
@@ -1,38 +1,28 @@
...
-        if (Integer.parseInt(sequence[i]) < Integer.parseInt(sequence[i - 1])) {
-          System.out.println("Your numerical sequence is not nondecreasing.");
-          return;
-        } 
-      } 
-      System.out.println("Your numerical sequence is increasing.");   
-    } catch (NumberFormatException e) {
-      System.err.println("You've entered invalid number format.");
-    }    
+  static void output(boolean flag) {
```
Comment (AlexanderSoroka) [inline IncreasingSequence.java]: why static?

File: SequenceAnalysis.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Analyzes the entered sequence.
+ */
+public class SequenceAnalysis {
+  /**
+   * Determines the kind of sequence.
+   *
+   * @param sequence a user-defined sequence of numbers.
+   * @return true if our sequence is increasing.
```
Comment (AlexanderSoroka) [inline SequenceAnalysis.java]: There is should be only one return

File: IncreasingSequence.java [hunk 9]
```diff
@@ -1,38 +1,24 @@
-import java.util.Scanner;
- 
 public class IncreasingSequence {
   public static void main(String[] args) {
-    if (args.length != 0) {
-      nondecreasing(args);
+    SequenceAnalysis sequence = new SequenceAnalysis();
+    Verification sequenceVerification = new Verification();
+    Output out = new Output();
+    if (args.length != 0 && !sequenceVerification.verification(args)) {
```
Comment (AlexanderSoroka) [inline IncreasingSequence.java]: that looks weird - analyze only if args.length > 0 and only if it doesn't pass verification....

File: IncreasingSequence.java [hunk 10]
```diff
@@ -1,38 +1,24 @@
...
+        out.output(args);
     } else {
-      Scanner scan = new Scanner(System.in);
-      System.out.print("Enter the numbers through a space: ");
-      String sequence = scan.nextLine();
-      String sequenceParts[] = sequence.split(" ");
-      nondecreasing(sequenceParts);
+      ReadingFromTheConsole read = new ReadingFromTheConsole();
+      String[] sequenceReading = read.reading();
+      if (!sequenceVerification.verification(sequenceReading)) {
```
Comment (AlexanderSoroka) [inline IncreasingSequence.java]: again, analyze sequence only if it doesn't pass verification

File: IncreasingSequence.java [hunk 11]
```diff
@@ -1,38 +1,24 @@
...
+        sequence.sequenceAnalysis(sequenceReading);
+        out.output(sequenceReading);
+      } else {  
+        do {
+          sequenceReading = read.reading();
+        } while (sequenceVerification.verification(sequenceReading));
```
Comment (AlexanderSoroka) [inline IncreasingSequence.java]: Repeat if sequence has been verified? The weirdest thing I saw

File: OutputMessages.java [hunk 12]
```diff
@@ -0,0 +1,27 @@
+/**
+ * Displays information on the screen about the type of sequence.
+ */
+public class OutputMessages {
+  /**
+   * Displays messages on the screen about the type of sequence.
+   * 
+   * @param flag Talks about the kind of sequence.  
+   */
+  private void messages(boolean flag) {
```
Comment (AlexanderSoroka) [inline OutputMessages.java]: non-appropriate mehtod name

File: OutputMessages.java [hunk 13]
```diff
@@ -0,0 +1,27 @@
+/**
+ * Displays information on the screen about the type of sequence.
+ */
+public class OutputMessages {
+  /**
+   * Displays messages on the screen about the type of sequence.
```
Comment (AlexanderSoroka) [inline OutputMessages.java]: wrong comment

File: OutputMessages.java [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+   */
+  private void messages(boolean flag) {
+    if (!flag) {
+      System.out.println("Your numerical sequence is not nondecreasing.");
+    } else {
+      System.out.println("Your numerical sequence is increasing.");
+    }
+  }
+  /**
+   * Calls the private method messages.
```
Comment (AlexanderSoroka) [inline OutputMessages.java]: wrong comment

File: OutputMessages.java [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+   *
+   * @param sequence our sequence.
+   */
+  public void output(String[] sequence) {
+    SequenceAnalysis analysis = new SequenceAnalysis();
```
Comment (AlexanderSoroka) [inline OutputMessages.java]: single responsibility violation

File: SequenceAnalysis.java [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+public class SequenceAnalysis { 
+  /* Analyzes our sequence.
+   * @param our sequence.
+   * @return kindSequence if true -- increasing sequence.
+   */
+  public boolean sequenceAnalysis(String[] sequence) {
+    boolean kindSequence = false;
+    for (int i = 0; i < sequence.length - 1; i++) {
+      if (Integer.parseInt(sequence[i]) > Integer.parseInt(sequence[i + 1])) {
+        kindSequence = false;
```
Comment (AlexanderSoroka) [inline SequenceAnalysis.java]: return false;

File: IncreasingSequence.java (hunk 11 above)
Comment (VadimBelkin) [inline IncreasingSequence.java]: Repeat until it becomes true.

File: IncreasingSequence.java (hunk 9 above)
Comment (VadimBelkin) [inline IncreasingSequence.java]: I understund. On the contrary.

File: OutputMessages.java (hunk 15 above)
Comment (VadimBelkin) [inline OutputMessages.java]: I don't know, as do, to method output was private. And at the same time, in order for single responsibility.

File: task_DEV-6/OutputMessagesAboutKindSequence.java [hunk 17]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Determines which type of sequence should be in the message 
+ * and displays it on the screen.
+ */
+public class OutputMessagesAboutKindSequence {
+  boolean kindSequence = false;
```
Comment (AlexanderSoroka) [inline task_DEV-6/OutputMessagesAboutKindSequence.java]: unused field

File: task_DEV-6/OutputMessagesAboutKindSequence.java [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+  /**
+   * Determines which type of sequence should be in the message.
+   * and displays it on the screen.
+   *
+   * @param flag Talks about the kind of sequence.  
+   */
+  public void outputmessageAboutKindSequence(boolean flag) {
```
Comment (AlexanderSoroka) [inline task_DEV-6/OutputMessagesAboutKindSequence.java]: codestyle violation

Comment (AlexanderSoroka) [inline task_DEV-6/OutputMessagesAboutKindSequence.java]: non-appropriate argument name

File: task_DEV-6/ReadingFromTheConsole.java [hunk 19]
```diff
@@ -0,0 +1,18 @@
+import java.util.Scanner;
+/**
+ * Reads a numeric sequence from the console.
+ */
+public class ReadingFromTheConsole {
+  /**
+   * Asks to enter.
```
Comment (AlexanderSoroka) [inline task_DEV-6/ReadingFromTheConsole.java]: non-appropriate comment

File: task_DEV-6/IncreasingSequence.java [hunk 20]
```diff
@@ -0,0 +1,21 @@
+public class IncreasingSequence {
+  public static void main(String[] args) {
+    SequenceAnalysis sequence = new SequenceAnalysis();
+    Verification sequenceVerification = new Verification();
+    OutputMessagesAboutKindSequence out = new OutputMessagesAboutKindSequence();
+    if (args.length != 0 && sequenceVerification.verification(args)) {
```
Comment (AlexanderSoroka) [inline task_DEV-6/IncreasingSequence.java]: it's hard to read and understand all these nested ifs and cycle... please split it to separate methods

File: task_DEV-6/SequenceAnalysis.java [hunk 21]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Analyzes our sequence.
+ */
+public class SequenceAnalysis { 
+  /* Analyzes our sequence.
+   * @param our sequence.
+   * @return kindSequence if true -- increasing sequence.
```
Comment (AlexanderSoroka) [inline task_DEV-6/SequenceAnalysis.java]: not all return value options are described

File: task_DEV-6/SequenceAnalysis.java [hunk 22]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Analyzes our sequence.
+ */
+public class SequenceAnalysis { 
+  /* Analyzes our sequence.
```
Comment (AlexanderSoroka) [inline task_DEV-6/SequenceAnalysis.java]: javadoc violation

File: task_DEV-6/Verification.java [hunk 23]
```diff
@@ -0,0 +1,24 @@
...
+   */
+  public boolean verification(String[] sequence) {
+    boolean repeat = false;
+    try {
+      for (int i = 0; i < sequence.length - 1; i++) {
+        Integer.parseInt(sequence[i]);
+      }
+      repeat = true; 
+    } catch (NumberFormatException e) {
+      System.err.println("You've entered invalid number format.");
```
Comment (AlexanderSoroka) [inline task_DEV-6/Verification.java]: single responsibility violation

File: task_DEV-6/Verification.java [hunk 24]
```diff
@@ -0,0 +1,24 @@
+/**
+ * Verifications our sequence.
+ */
+public class Verification {
+  /**
+   * Verifications our sequence.
+   *
+   * @param sequence our sequence.
+   * @return repeat if true -- the exception is caught.
```
Comment (AlexanderSoroka) [inline task_DEV-6/Verification.java]: "the exception is caught" and what?

File: task_DEV-6/OutputMessagesAboutKindSequence.java [hunk 25]
```diff
@@ -0,0 +1,19 @@
...
+ * and displays it on the screen.
+ */
+public class OutputMessagesAboutKindSequence {
+  /**
+   * Determines which type of sequence should be in the message.
+   * and displays it on the screen.
+   *
+   * @param kindSequence Talks about the kind of sequence.  
+   */
+  public void outputMessageAboutKindSequence(boolean kindSequence) {
```
Comment (AlexanderSoroka) [inline task_DEV-6/OutputMessagesAboutKindSequence.java]: non-appropriate argument name

File: task_DEV-6/ReadingFromTheConsole.java [hunk 26]
```diff
@@ -0,0 +1,17 @@
+import java.util.Scanner;
+/**
+ * Reads a numeric sequence from the console.
+ */
+public class ReadingFromTheConsole {
+  /**
+   * Reads a numeric sequence from the console.
+   *
+   * @return sequenceParts the entered sequence.
```
Comment (AlexanderSoroka) [inline task_DEV-6/ReadingFromTheConsole.java]: what does sequenceParts means?

File: task_DEV-6/SequenceAnalysis.java [hunk 27]
```diff
@@ -0,0 +1,19 @@
+/**
+ * Analyzes our sequence.
+ */
+public class SequenceAnalysis { 
+  /** Analyzes our sequence.
```
Comment (AlexanderSoroka) [inline task_DEV-6/SequenceAnalysis.java]: javadoc violation

File: task_DEV-6/SequenceAnalysis.java [hunk 28]
```diff
@@ -0,0 +1,19 @@
...
+   * @param our sequence.
+   * @return kindSequence if true -- increasing sequence.
```
Comment (AlexanderSoroka) [inline task_DEV-6/SequenceAnalysis.java]: what does kindSequence means?

File: task_DEV-6/SequenceAnalysis.java [hunk 29]
```diff
@@ -0,0 +1,19 @@
+/**
+ * Analyzes our sequence.
+ */
+public class SequenceAnalysis { 
+  /** Analyzes our sequence.
+   * @param our sequence.
```
Comment (AlexanderSoroka) [inline task_DEV-6/SequenceAnalysis.java]: what does our means?

File: task_DEV-6/SequenceAnalysis.java [hunk 30]
```diff
@@ -0,0 +1,19 @@
...
+   * @return kindSequence if true -- increasing sequence.
+   *                      If false -- not non-decreasing sequence.
+   */
+  public boolean sequenceAnalysis(String[] sequence) {
+    boolean kindSequence = false;
```
Comment (AlexanderSoroka) [inline task_DEV-6/SequenceAnalysis.java]: unused variable

File: task_DEV-6/Verification.java [hunk 31]
```diff
@@ -0,0 +1,51 @@
+/**
+ * Verifications our sequence.
+ */
+public class Verification {
+  /**
+   * Checks the format.
+   *
+   * @param sequence our sequence.
+   * @return repeat if true -- format is true.
```
Comment (AlexanderSoroka) [inline task_DEV-6/Verification.java]: repeat if true? What does it mean? Repeat what?

File: task_DEV-6/Verification.java [hunk 32]
```diff
@@ -0,0 +1,51 @@
...
+  /**
+   * Checks the format.
+   *
+   * @param sequence our sequence.
+   * @return repeat if true -- format is true. 
+   *                else -- format is false.
+   *                Exception will be processed in main. 
+   */
+  public boolean checksTheFormat(String[] sequence) {
+    boolean repeat = false;
```
Comment (AlexanderSoroka) [inline task_DEV-6/Verification.java]: avoid using flags

File: task_DEV-6/Verification.java [hunk 33]
```diff
@@ -0,0 +1,51 @@
...
+      for (int i = 0; i < sequence.length - 1; i++) {
+        Integer.parseInt(sequence[i]);
+      }
+      repeat = true; 
+    } catch (NumberFormatException e) {
+      repeat = false;
+    }
+    return repeat;
+  }
+  /**
```
Comment (AlexanderSoroka) [inline task_DEV-6/Verification.java]: blank line before javadoc comment missed

File: task_DEV-6/Verification.java [hunk 34]
```diff
@@ -0,0 +1,51 @@
...
+   * Checks the length our sequence.
+   *
+   * @param sequence our sequence.
+   * @return true, if string is entered. 
+   *         false, if length of the string <= 1. 
+   */
+  public boolean checksTheLength(String[] sequence) {
+    if (sequence.length - 1  >= 1) {
```
Comment (AlexanderSoroka) [inline task_DEV-6/Verification.java]: `return sequence.length - 1  >= 1;`

File: task_DEV-6/Verification.java [hunk 35]
```diff
@@ -0,0 +1,51 @@
...
+  }
+  /**
+   * Verifications our sequence.
+   *
+   * @param sequence our sequence.
+   * @return true, if input is correct.
+   *         false, if input isn't correct.
+   */ 
+  public boolean verification(String[] sequence) {
+    if (checksTheLength(sequence) && checksTheFormat(sequence)) {
```
Comment (AlexanderSoroka) [inline task_DEV-6/Verification.java]: `return checksTheLength(sequence) && checksTheFormat(sequence);`

**Pull request merged**